devel/lazarus/Portfile

```diff
@@ -0,0 +1,131 @@
+# with osprinters.pas the line number needs to be given
+# Note: I did not manage a solution with reinplace
+    system -W ${worksrcpath}/components/printers \
+        "sed -i.tmp '50 s|LCLGtk2|LCLnogui\}\{\$I cupsprinters_h.inc\}\{\$ENDIF\}\{\$IFDEF LCLGtk2|g' osprinters.pas && \
+         sed -i.tmp '87 s|LCLGtk2|LCLnogui\}\{\$I cupsprinters.inc\}\{\$ENDIF\}\{\$IFDEF LCLGtk2|g'   osprinters.pas && \
+         sed -i.tmp 's|LCLGtk2|LCLnogui\}uses udlgSelectPrinter, udlgPropertiesPrinter, udlgPageSetup;\{\$I cupsprndialogs.inc\}\{\$ENDIF\}\{\$IFDEF LCLGtk2|g' ${worksrcpath}/components/printers/printersdlgs.pp "
+
+# some more fixes for nogui taking cocoa as a template
+# this may break, as soon as cocoa gets working versions
+    copy       ${worksrcpath}/components/lclextensions/include/cocoa ${worksrcpath}/components/lclextensions/include/nogui
+    reinplace "s|CocoaInt, CocoaGDIObjects, ||g"                     ${worksrcpath}/components/lclextensions/include/nogui/uses.inc
+    reinplace "s|Cocoa Interface|NoGui Interface|g"                  ${worksrcpath}/components/lclextensions/include/nogui/delphicompat.inc
+
+# Note: I did not manage a solution with reinplace
+    system -W ${worksrcpath}/components/lclextensions/include/nogui \
+        "sed -i.tmp 's|\$define HAS_GETCURRENTOBJECT|.\$define HAS_GETCURRENTOBJECT|g' delphicompat.inc && \
+         sed -i.tmp '21,27d' delphicompat.inc && \
+         sed -i.tmp '46,68d' delphicompat.inc "
+
+    copy ${worksrcpath}/components/virtualtreeview/units/cocoa         ${worksrcpath}/components/virtualtreeview/units/nogui
+    copy ${worksrcpath}/components/virtualtreeview/include/intf/cocoa/ ${worksrcpath}/components/virtualtreeview/include/intf/nogui
+
+}
+
+build.target       bigide
+build.post_args    OPT=-gl LCL_PLATFORM=cocoa
+use_parallel_build no
+
+destroot {
+    xinstall -m 755 -d ${destroot}${prefix}/etc/lazarus
+    xinstall -m 644 ${worksrcpath}/tools/install/macosx/environmentoptions.xml ${destroot}${prefix}/etc/lazarus
+
+    system -W ${worksrcpath} "make install INSTALL_PREFIX=${destroot}${prefix}"
+
+# fix the symbolic links in ${destroot}${prefix}/bin
+    system -W ${destroot}${prefix}/bin \
+        "ln -sf ../share/lazarus/lazarus      lazarus-ide  && \
+         ln -sf ../share/lazarus/startlazarus startlazarus && \
+         ln -sf ../share/lazarus/lazbuild     lazbuild     "
+
+# fix the symbolic links in the application bundle
+    system -W ${destroot}${prefix}/share/lazarus/lazarus.app/Contents/MacOS \
+        "ln -sf ${prefix}/share/lazarus/lazarus      lazarus      && \
+         ln -sf ${prefix}/share/lazarus/startlazarus startlazarus "
+
+# remove the faulty app bundle in Resources
+    delete ${destroot}${prefix}/share/lazarus/lazarus.app/Contents/Resources/startlazarus.app
+
+# copy lazarus.app to MacPorts application directory
+    copy ${destroot}${prefix}/share/lazarus/lazarus.app ${destroot}${applications_dir}
```

Comment:
I can't help but assume we have a way of dealing with .app bundles already.

Reply:
Maybe, I had the wrong example them. Hints to the docs or an existing Portfile?

Reply:
@mf2k may know. I am pretty sure we have some machinery because of the existence of the `applications_dir` variable documented in the MacPorts Guide.

Reply:
Unfortunately, I don't know either way.

Reply:
I don't think we have a way of "dealing with .app bundles". Apps written for macOS should already have a build system that creates proper .app bundles. For programs that behave somewhat reasonably to being presented as an .app bundle yet don't contain code to create them (programs that use libsdl are a good example), the app portgroup can be used.

The commands I see above that move things around for the .app bundle look unusual, as if this build system tries to create an .app bundle and doesn't do a good job of it. If that's the case, then fixing it in the portfile is fine for now, though ideally please fix the build system and submit the fixes to upstream.

I'm unclear why the app is being copied into applications_dir, rather than being moved there. Why would we want to install two copies of it?

Reply:
I'll report upstream. copy is replaced by move.